backends/p4tools/cmake/FindZ3.cmake

```diff
@@ -0,0 +1,68 @@
+find_package(PkgConfig)
+pkg_check_modules(PC_Z3 QUIET Z3)
+
+find_path(
+  Z3_INCLUDE_DIR
+  NAMES z3++.h
+  HINTS ${PC_Z3_INCLUDEDIR} ${PC_Z3_INCLUDE_DIRS}
+  PATH_SUFFIXES z3 Z3
+)
+
+find_library(
+  Z3_LIBRARY
+  NAMES libz3.a
```

Comment:
The following change would allow to build with `libz3.so` (e.g., when installed from a package)
```Suggestion
  NAMES z3
```